Sources/Workspace/Workspace.swift

```diff
@@ -634,6 +635,12 @@ extension Workspace {
         activeResolver = nil
 
         guard !diagnostics.hasErrors else { return }
+        
+        guard !dryRun else {
+            let changes = diagnostics.wrap { return try computePackageStateChanges(root: graphRoot, resolvedDependencies: updateResults, updateBranches: true) } ?? []
+            logPackageChanges(changes: changes, pins: pinsStore)
```

Comment:
`Workspace` is a high-level library so it'll be better if we return a structured object from here. Clients can then do whatever they want with the object. In this particular case, maybe you can accept an `inout` array to fill for dry run.

Reply:
Great idea! I think it looks way better than splitting `updateDependencies`  into `updateDependencies` and `updateDependenciesDryRun`